embedded-graphics/src/drawable.rs

```diff
@@ -107,16 +66,60 @@ where
     C: PixelColor,
 {
     /// Draw the graphics object using the supplied DrawTarget.
-    fn draw<T: DrawTarget<C>>(self, display: &mut T);
+    fn draw<T: DrawTarget<C>>(self, display: &mut T) -> Result<(), T::Error>;
+}
+
+/// A single pixel.
+///
+/// `Pixel` objects are used to specify the position and color of drawn pixels.
+///
+/// # Examples
+///
+/// The [`Drawable`] trait is implemented for `Pixel` which allows single pixels
+/// to be drawn to a [`DrawTarget`]:
+/// ```
+/// use embedded_graphics::{pixelcolor::BinaryColor, prelude::*};
+/// # use embedded_graphics::mock_display::MockDisplay;
+/// # let mut display = MockDisplay::new();
+///
+/// Pixel(Point::new(1, 2), BinaryColor::On).draw(&mut display)?;
+/// # Ok::<(), core::convert::Infallible>(())
+/// ```
+///
+/// Iterators with `Pixel` items can also be drawn:
+///
+/// ```
+/// use embedded_graphics::{pixelcolor::BinaryColor, prelude::*};
+/// # use embedded_graphics::mock_display::MockDisplay;
+/// # let mut display = MockDisplay::new();
+///
+/// (0..100)
+///     .map(|i| Pixel(Point::new(i, i * 2), BinaryColor::On))
+///     .draw(&mut display)?;
+/// # Ok::<(), core::convert::Infallible>(())
+/// ```
+///
+/// [`Drawable`]: trait.Drawable.html
+/// [`DrawTarget`]: ../trait.DrawTarget.html
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct Pixel<C: PixelColor>(pub Point, pub C);
+
+impl<C> Drawable<C> for Pixel<C>
+where
+    C: PixelColor,
+{
+    fn draw<T: DrawTarget<C>>(self, display: &mut T) -> Result<(), T::Error> {
```

Comment:
The type parameter should always be called the same:
```suggestion
    fn draw<D: DrawTarget<C>>(self, display: &mut D) -> Result<(), D::Error> {
```